eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Bbody.hs

```diff
@@ -142,27 +133,81 @@ instance
   ) =>
   EncCBOR (ConwayBbodyPredFailure era)
   where
-  encCBOR =
-    encode . \case
-      WrongBlockBodySizeBBODY x y -> Sum WrongBlockBodySizeBBODY 0 !> To x !> To y
-      InvalidBodyHashBBODY x y -> Sum (InvalidBodyHashBBODY @era) 1 !> To x !> To y
-      LedgersFailure x -> Sum (LedgersFailure @era) 2 !> To x
-      TooManyExUnits x y -> Sum TooManyExUnits 3 !> To x !> To y
-      BodyRefScriptsSizeTooBig x y -> Sum BodyRefScriptsSizeTooBig 4 !> To x !> To y
+  encCBOR pf = ifEncodingVersionAtLeast (natVersion @11) (elevenOnwards pf) (beforeEleven pf)
+    where
+      elevenOnwards =
+        encode . \case
+          WrongBlockBodySizeBBODY mm -> Sum WrongBlockBodySizeBBODY 0 !> To mm
+          InvalidBodyHashBBODY mm -> Sum (InvalidBodyHashBBODY @era) 1 !> To mm
+          LedgersFailure x -> Sum (LedgersFailure @era) 2 !> To x
+          TooManyExUnits mm -> Sum TooManyExUnits 3 !> To mm
+          BodyRefScriptsSizeTooBig mm -> Sum BodyRefScriptsSizeTooBig 4 !> To mm
+      beforeEleven =
+        encode . \case
+          WrongBlockBodySizeBBODY mm@(Mismatch _ _) ->
+            Sum WrongBlockBodySizeBBODY 0
+              !> E (\(Mismatch {..}) -> encCBOR mismatchSupplied <> encCBOR mismatchExpected) mm
+          InvalidBodyHashBBODY mm@(Mismatch _ _) ->
+            Sum (InvalidBodyHashBBODY @era) 1
+              !> E (\(Mismatch {..}) -> encCBOR mismatchSupplied <> encCBOR mismatchExpected) mm
+          LedgersFailure x -> Sum (LedgersFailure @era) 2 !> To x
+          TooManyExUnits mm@(Mismatch _ _) ->
+            Sum TooManyExUnits 3
+              !> E (\(Mismatch {..}) -> encCBOR mismatchSupplied <> encCBOR mismatchExpected) mm
+          BodyRefScriptsSizeTooBig mm@(Mismatch _ _) ->
+            Sum BodyRefScriptsSizeTooBig 4
+              !> E (\(Mismatch {..}) -> encCBOR mismatchSupplied <> encCBOR mismatchExpected) mm
```

Comment:
Same as with decoder, protocol version should not be used here. We really need to preserve the serialization for Conway.

Furthermore, if you add `Enc/DecCBORGRoup` instances for `Mismatch`, then this becomes a simple:
```suggestion
  encCBOR =
    encode . \case
      WrongBlockBodySizeBBODY m -> Sum WrongBlockBodySizeBBODY 0 !> E encCBORGRoup m
      InvalidBodyHashBBODY m -> Sum (InvalidBodyHashBBODY @era) 1 !> E encCBORGRoup m
      LedgersFailure x -> Sum (LedgersFailure @era) 2 !> To x
      TooManyExUnits m -> Sum TooManyExUnits 3 !> E encCBORGRoup m
      BodyRefScriptsSizeTooBig m -> Sum BodyRefScriptsSizeTooBig 4 !> E encCBORGRoup m
```

Reply:
With `E`, the length of the list ended up being wrong because the two fields got inlined but counted only one element. I have added `{To|From}Group` to the `Coders` to take into account `{Enc|Dec}CBORGroup`-based (de)serializations based on ideas from @teodanciu  and @TimSheard. Let me know what you think 😃 .